src/Agent.Worker/TestResults/ResultsCommandExtension.cs

```diff
@@ -260,23 +309,27 @@ private PublishOptions GetPublishOptions()
         }
 
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Maintainability", "CA2000:Dispose objects before losing scope", MessageId = "connection")]
-        private async Task PublishTestRunDataAsync(String teamProject, TestRunContext testRunContext)
+        private async Task PublishTestRunDataAsync(string teamProject, TestRunContext testRunContext)
         {
             bool isTestRunOutcomeFailed = false;
 
-            var connection = WorkerUtilities.GetVssConnection(_executionContext);
-            var featureFlagService = _executionContext.GetHostContext().GetService<IFeatureFlagService>();
-            featureFlagService.InitializeFeatureService(_executionContext, connection);
-            var publishTestResultsLibFeatureState = featureFlagService.GetFeatureFlagState(TestResultsConstants.UsePublishTestResultsLibFeatureFlag, TestResultsConstants.TFSServiceInstanceGuid);
             _telemetryProperties.Add("UsePublishTestResultsLib", publishTestResultsLibFeatureState);
+            var connection = WorkerUtilities.GetVssConnection(_executionContext);
```

Comment:
Using? VssConnection implements IDisposable

Reply:
Follow up PR for implementing using statement for the connection to dispose the connection post completion of the task.